unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/nio/channels/FileChannelTest.scala

```diff
@@ -296,4 +297,54 @@ class FileChannelTest {
       }
     }
   }
+
+  @Test def mapMethodIsTidy(): Unit = {
+    withTemporaryDirectory { dir =>
+      val data = s"abcdef"
+      val dataBytes = data.getBytes("UTF-8")
+
+      val f = dir.resolve("mapArguments.txt")
+      Files.write(f, dataBytes)
+
+      val lines = Files.readAllLines(f)
+      assertEquals("lines size", 1, lines.size())
+      assertEquals("lines content", data, lines.get(0))
+
+      val channel = FileChannel.open(
+        f,
+        StandardOpenOption.READ,
+        StandardOpenOption.WRITE
+      )
+
+      try {
```

Comment:
You could use `Using` to manage the resource here rather than `try`, `finally`. https://www.scala-lang.org/api/2.13.6/scala/util/Using$.html

Reply:
Eric,  
Thank you for the suggestion.  This code needs to also run on Scala 2.12.n.   
I believe that `Using` is not available for that Scala version. `Tis a pity, 
the `Using` idiom is much closer to the Java `try-with-resources` idiom.
Did I miss something in Scala 2.12?